oak-segment-azure/src/main/java/org/apache/jackrabbit/oak/segment/azure/AzureRepositoryLock.java

```diff
@@ -100,7 +105,11 @@ private void refreshLease() {
             try {
                 long timeSinceLastUpdate = (System.currentTimeMillis() - lastUpdate) / 1000;
                 if (timeSinceLastUpdate > INTERVAL / 2) {
+                    writeAccessController.disableWriting();
```

Comment:
What about the performance impact of this change?
My concern is we are blocking writes at every renewal, even though the lease has not expired and writes are still safe. Even a successful call to `blob.acquireLease()` could take a long time if Azure is slow.

I would suggest to introduce a RenewalDeadiine timeout, while increasing the rate of renewals: for example, if lease duration is 60 seconds, we should renew the lease every 5 seconds, but if after a deadline of 40 seconds we couldn't renew the lease, then and only then we block the writes.

Reply:
> Even a successful call to blob.acquireLease() could take a long time if Azure is slow.

It is possible to do that optimization, but if Azure, as you said, is slow, then threads in SegmentWriteQueue will be slow to write to the remote repo as well. 

If we block writes only 40 seconds after issuing blob.renewLease, the lease at that time will be 5 + 40 seconds old. 

The writer thread might pass ```WriteAccessController#checkWritingAllowed```condition at 44 seconds after lease renewal and then struggle for the next 20 seconds to write the segment.

Reply:
5 and 40 seconds were just examples and could be adjusted to, say, 10 and 30.
With those values, with my proposal, we block writes after lease renewal fails multiple times (after first renewal failure with a OperationTimeout, retry happens immediately) but at most after 30 seconds.
Currently we renew only after 30 seconds, which seems too late to recover from safely, that's why I argue that we should renew more often and have a separate deadline timeout to block writes way before the lease expires.

Also see my other comment, where I argue we should block `SegmentWriteQueue` as well, to prevent the scenario you mention of a writer thread still trying to write because it just passed the check before the block.